### ISSUE TITLE
GatewayInternalError: Error while paying invoice, for every payment

### ISSUE THREAD

**Issue by TonyGiorgio** [fedimint/fedimint#4943]
I have a feeling something more is happening here than a simple `pay_invoice` failure, but it's hard to tell by the logs returned: 

```
8ea0 2024-04-12 16:05:48.983 TRACE [mutiny_core::federation:901] Streamed Outcome received: WaitingForRefund { block_height: 839382, gateway_error: GatewayInternalError { error_code: Some(400), error_message: "Error while paying lightning invoice. Outgoing contract will be refunded." } }
8ea0 2024-04-12 16:05:49.075 TRACE [mutiny_core::federation:901] Streamed Outcome received: Refunded { gateway_error: GatewayInternalError { error_code: Some(400), error_message: "Error while paying lightning invoice. Outgoing contract will be refunded." } }
8ea0 2024-04-12 16:05:49.075 TRACE [mutiny_core::federation:905] Streamed Outcome final, returning
8ea0 2024-04-12 16:05:49.075 TRACE [mutiny_core::federation:917] Done with stream outcome, status: Failed
```

The federation/gateway in question is for the atlantis federation, however, on my own device, payments are working fine. We have two users with this problem: https://github.com/MutinyWallet/mutiny-web/issues/1041

Is there anything else we can do to troubleshoot this? Maybe more detailed errors we can return back from the gateway, or even better if there's any insights from this gateway if anyone can check on it.

**Comment by TonyGiorgio** [comment]
@elsirion mentioned something about clock time could be incorrect? I'm not sure actually if that would be the case, it seems like it was working for them at some point.

**Comment by TonyGiorgio** [comment]
This was due to a gateway that changed fees. I was not aware that we needed to call `update_gateway_cache` at all. I have two suggestions: 
1. Have the gateway cache update automatically
2. Return an error indicating a fee problem, not just a normal payment problem.

**Comment by elsirion** [comment]
It was mentioned that the problem was likely HODL-invoice related. There might have been a regression since apparently we implemented support in #3615, needs further investigation.

**Comment by TonyGiorgio** [comment]
> It was mentioned that the problem was likely HODL-invoice related. There might have been a regression since apparently we implemented support in #3615, needs further investigation.

I have heard other issues around hodl invoices but have not been able to reproduce it. Payments to zeus lightning addresses (which are hodl) seem to work for me. Not sure if it's after a long time or not.

**Comment by justinmoon** [comment]
dev call: gateway should have a mode where it gives more verbose error messages.
